### ISSUE TITLE
some crypto module tests don't pass on Ubuntu 18.04

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#9794]
library/packages/Crypto/saru/blowfish/bfTest1.chpl
library/packages/Crypto/saru/hashing/hash1.chpl

I get errors along the lines of:
```
In file included from /tmp/chpl-mppf-17728.deleteme/_main.c:46:0:
/tmp/chpl-mppf-17728.deleteme/Crypto.c: In function 'bfEncrypt_chpl':
/tmp/chpl-mppf-17728.deleteme/Crypto.c:615:16: error: storage size of 'ctx_chpl' isn't known
 EVP_CIPHER_CTX ctx_chpl;
                ^~~~~~~~

```

on Ubuntu 18.04 with GCC 7.3 and openssl 1.1.0g.

It seems that EVP_CIPHER_CTX is defined as an opaque type in ossl_type.h:
``` C
typedef struct evp_cipher_ctx_st EVP_CIPHER_CTX;
```

and this S.O. answer points out that "EVP_CIPHER_CTX was made opaque in OpenSSL 1.1.0" and discusses what to do about it:
https://stackoverflow.com/questions/26345175/correct-way-to-free-allocate-the-context-in-the-openssl

**Comment by mppf** [comment]
@saru95 - are you able to have a look at this OpenSSL 1.1.0 issue? Thanks.

**Comment by mppf** [comment]
@saru95 - have you had a chance to look at this ? I believe the fix will be relatively straightforward. Thanks.

**Comment by saru95** [comment]
Please do a `brew install openssl@1.1` if on macOS. Homebrew installs `1.0.2` by default. Also, `1.1` is not backward compatible and there are lots of breaking changes between the 2 releases. I'm thinking of adding a note to the docs stating that the lib supports openssl >1.1. What do you think?

Also, the `new` and `free` API is available since 1.0 but their digest variants have been introduced in `1.1`. This is so frustrating. 

PR: https://github.com/chapel-lang/chapel/pull/10601

**Comment by mppf** [comment]
@saru95 - Requiring 1.1 is OK with me but if you want to also support 1.0.2 for a while, you could add some macros in modules/packages/CryptoHandlers/rsa_complex_bypass_handler.h .
